### PR TITLE
Quark-Engine integration

### PR DESCRIPTION
**Pull request by pulorsok**
#1119
Hi @skylot, 
This is a quick PoC for quark integration, a few features have been done:
1. add a quark button on Toolbar
2. click to execute the Quark command
3. output quark report as JSON
4. show report on Tabbed panel

However, there is still some feature have not completed, such as code navigation by click on malware behaviour in the report. And also, Jadx doesn't have plugin framework yet, so I implemented quark integration into jadx-gui code base.

If you want to integrate quark by using plugin, we can help to implement some plugin feature for contribution.

Please let me know if you have any suggestion or question. :D

### CODE REVIEW

**Review by skylot — APPROVED** [review]

File: jadx-gui/src/main/java/jadx/gui/ui/QuarkDialog.java [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+				String outputPath = quarkReportFile.getAbsolutePath().toString();
+				String apkName = selectFile.getSelectedItem().toString();
+				String apkPath = null;
+				for (Path path : files) {
+					if (path.getFileName().toString().equals(apkName)) {
+						apkPath = path.toString();
+					}
+				}
+				String cmd = "quark -a " + apkPath + " -s -o " + outputPath;
```
Comment (bagipro) [inline jadx-gui/src/main/java/jadx/gui/ui/QuarkDialog.java:207]: @pulorsok 
Potential command injection. Please use String array to prevent this

Comment (skylot) [reply jadx-gui/src/main/java/jadx/gui/ui/QuarkDialog.java:207]: @bagipro thanks for check, but unfortunately I already merged this PR :disappointed: 
I will fix these myself, do you have any links how to do it correctly? Only replacing with array is enough?

Comment (bagipro) [reply jadx-gui/src/main/java/jadx/gui/ui/QuarkDialog.java:207]: @skylot 
Yeah, the array is enough

Comment (skylot) [reply jadx-gui/src/main/java/jadx/gui/ui/QuarkDialog.java:207]: @bagipro thanks! Will fix ASAP

Comment (skylot) [reply jadx-gui/src/main/java/jadx/gui/ui/QuarkDialog.java:207]: Fixed in 52412dfe31a1184a512636349cbe3a68792ecb88

**Pull request merged**